pkg/scanners/scannerv4/nodescan_convert.go

```diff
@@ -9,144 +9,79 @@ import (
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/protocompat"
 	"github.com/stackrox/rox/pkg/set"
-	"github.com/stackrox/rox/pkg/utils"
 )
 
-var (
-	rhcosOSImageRegexp = regexp.MustCompile(`(Red Hat Enterprise Linux) (CoreOS) ([\d])([\d]+)`)
-)
-
-const (
-	rhcosFullName = "Red Hat Enterprise Linux CoreOS"
-)
-
-func toNodeScan(r *v4.VulnerabilityReport, osImageRef string) *storage.NodeScan {
-	// TODO(ROX-26593): Instead of fixing notes here, add RHCOS DistributionScanner to ClairCore
-	fixedNotes := fixNotes(toStorageNotes(r.Notes), osImageRef)
+const rhcosFullName = "Red Hat Enterprise Linux CoreOS"
 
-	convertedOS := toOperatingSystem(osImageRef)
-	if convertedOS == "" {
-		log.Warnf("Could not determine operating system from OSimage ref %s", osImageRef)
-	}
+var rhcosOSImagePattern = regexp.MustCompile(`(Red Hat Enterprise Linux) (CoreOS) (\d)(\d+)`)
 
+func nodeScan(osImage string, vr *v4.VulnerabilityReport) *storage.NodeScan {
 	return &storage.NodeScan{
-		ScanTime:        protocompat.TimestampNow(),
-		Components:      toStorageComponents(r),
-		Notes:           fixedNotes,
 		ScannerVersion:  storage.NodeScan_SCANNER_V4,
-		OperatingSystem: convertedOS,
+		ScanTime:        protocompat.TimestampNow(),
+		OperatingSystem: nodeOS(osImage),
+		Components:      nodeComponents(vr),
+		Notes:           nodeNotes(vr, osImage),
 	}
 }
 
-func toOperatingSystem(ref string) string {
-	r := rhcosOSImageRegexp.FindStringSubmatch(ref)
+func nodeOS(osImage string) string {
+	r := rhcosOSImagePattern.FindStringSubmatch(osImage)
 	if len(r) != 5 {
-		return ""
+		return "unknown"
```

Comment:
This used to be the empty string, now it is "unknown", does the UI / any workflows that use this data know how to handle that?